src/nodes/conv.rs

```diff
@@ -154,18 +155,47 @@ pub fn attr_fs_to_fuse(path: &Path, inode: u64, attr: &fs::Metadata) -> fuse::Fi
     }
 }
 
+/// Converts a set of `flags` bitmask to an `fs::OpenOptions`.
+///
+/// `allow_writes` indicates whether the file to be opened supports writes or not.  If the flags
+/// don't match this condition, then this returns an error.
+pub fn flags_to_openoptions(flags: u32, allow_writes: bool) -> NodeResult<fs::OpenOptions> {
+    let flags = flags as i32;
+    let mut options = fs::OpenOptions::new();
+    options.read(true);
+    if flags & (fcntl::OFlag::O_WRONLY | fcntl::OFlag::O_RDWR).bits() != 0 {
```

Comment:
I feel like this would read better if you `fcntl::OFlag::from_bits_truncate` and then test `.contains(...)`.

Reply:
Done.